<a href="https://colab.research.google.com/github/mufeed0786/AI_CSV_Dashboard/blob/main/cancer_dataset_project.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Importing Librarries**

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

# **Loading csv_file**

In [2]:
cancer_patients =  pd.read_csv('/content/global_cancer_patients_2015_2024 .csv')

# Understanding **Data**

In [13]:
cancer_patients.head()
# Dataset contains 50,000 patient records with 15 features including demographics, risk factors, and cancer details.
# Genetic_Risk → rating (higher = higher risk)
# Survival_Years → rating (higher = better survival)

,Patient_ID,Age,Gender,Country_Region,Year,Genetic_Risk,Air_Pollution,Alcohol_Use,Smoking,Obesity_Level,Cancer_Type,Cancer_Stage,Treatment_Cost_USD,Survival_Years,Target_Severity_Score
0,PT0000000,71,Male,UK,2021,6.4,2.8,9.5,0.9,8.7,Lung,Stage III,62913.44,5.9,4.92
1,PT0000001,34,Male,China,2021,1.3,4.5,3.7,3.9,6.3,Leukemia,Stage 0,12573.41,4.7,4.65
2,PT0000002,80,Male,Pakistan,2023,7.4,7.9,2.4,4.7,0.1,Breast,Stage II,6984.33,7.1,5.84
3,PT0000003,40,Male,UK,2015,1.7,2.9,4.8,3.5,2.7,Colon,Stage I,67446.25,1.6,3.12
4,PT0000004,43,Female,Brazil,2017,5.1,2.8,2.3,6.7,0.5,Skin,Stage III,77977.12,2.9,3.62


In [ ]:
cancer_patients.tail()

,Patient_ID,Age,Gender,Country_Region,Year,Genetic_Risk,Air_Pollution,Alcohol_Use,Smoking,Obesity_Level,Cancer_Type,Cancer_Stage,Treatment_Cost_USD,Survival_Years,Target_Severity_Score
49995,PT0049995,80,Male,Pakistan,2023,2.3,7.5,2.8,3.8,2.9,Leukemia,Stage 0,71463.70,1.0,3.63
49996,PT0049996,40,Female,USA,2018,6.4,3.5,2.9,9.0,9.8,Colon,Stage I,49619.66,0.4,6.03
49997,PT0049997,74,Male,Pakistan,2015,6.2,1.6,8.7,4.7,4.0,Breast,Stage 0,5338.25,2.7,6.02
49998,PT0049998,21,Male,Brazil,2018,4.0,6.5,7.6,8.6,8.1,Liver,Stage IV,45100.47,7.5,6.54
49999,PT0049999,22,Male,Pakistan,2023,5.1,9.8,3.2,0.0,0.7,Liver,Stage 0,64023.82,7.4,3.76


In [ ]:
cancer_patients.shape
# Dataset contains 50,000 patient records with 15 features including demographics, risk factors, and cancer details.

(50000, 15)

In [ ]:
cancer_patients.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50000 entries, 0 to 49999
Data columns (total 15 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   Patient_ID             50000 non-null  object 
 1   Age                    50000 non-null  int64  
 2   Gender                 50000 non-null  object 
 3   Country_Region         50000 non-null  object 
 4   Year                   50000 non-null  int64  
 5   Genetic_Risk           50000 non-null  float64
 6   Air_Pollution          50000 non-null  float64
 7   Alcohol_Use            50000 non-null  float64
 8   Smoking                50000 non-null  float64
 9   Obesity_Level          50000 non-null  float64
 10  Cancer_Type            50000 non-null  object 
 11  Cancer_Stage           50000 non-null  object 
 12  Treatment_Cost_USD     50000 non-null  float64
 13  Survival_Years         50000 non-null  float64
 14  Target_Severity_Score  50000 non-null  float64
dtypes:

Looking at info of the dataset i see there is no null values .comprising most of the entries in float64 and object and 2 in int64.Memory usage of the data is 5.7+ MB

In [ ]:
cancer_patients.describe()

,Age,Year,Genetic_Risk,Air_Pollution,Alcohol_Use,Smoking,Obesity_Level,Treatment_Cost_USD,Survival_Years,Target_Severity_Score
count,50000.000000,50000.000000,50000.000000,50000.000000,50000.000000,50000.000000,50000.000000,50000.000000,50000.000000,50000.000000
mean,54.421540,2019.480520,5.001698,5.010126,5.010880,4.989826,4.991176,52467.298239,5.006462,4.951207
std,20.224451,2.871485,2.885773,2.888399,2.888769,2.881579,2.894504,27363.229379,2.883335,1.199677
min,20.000000,2015.000000,0.000000,0.000000,0.000000,0.000000,0.000000,5000.050000,0.000000,0.900000
25%,37.000000,2017.000000,2.500000,2.500000,2.500000,2.500000,2.500000,28686.225000,2.500000,4.120000
50%,54.000000,2019.000000,5.000000,5.000000,5.000000,5.000000,5.000000,52474.310000,5.000000,4.950000
75%,72.000000,2022.000000,7.500000,7.500000,7.500000,7.500000,7.500000,76232.720000,7.500000,5.780000
max,89.000000,2024.000000,10.000000,10.000000,10.000000,10.000000,10.000000,99999.840000,10.000000,9.160000


Looking at the stats of the data --
1.The average age of patients in the data is 54.421540,suggesting the cancer cases in the dataset are more among the middle age individuals.With min age 20 and max age 72.
2.The average year is around 2019, indicating that the dataset spans recent years..with maximum cases in 2024.
3.The average genetic risk score is 5, indicating a moderate level of genetic risk among patients.
4The average values of Air_Pollution and Alcohol_Use are also around the mid-range, indicating moderate exposure levels among patients.Smoking and Obesity levels are slightly lower compared to other risk factors.
5.The average treatment cost is $52467.29, with values ranging from $5000.050000 to $99999.85, indicating a wide variation in treatment expenses among patients.
6.The average survival score is 5, indicating a moderate level of treatment success among patients.
7.The average severity score is 4.95, suggesting that most cases fall under a moderate level of risk
8.The dataset shows that most risk factors and severity scores lie in the mid-range, indicating moderate overall risk levels among patients.

In [ ]:
cancer_patients.nunique()

,0
Patient_ID,50000
Age,70
Gender,3
Country_Region,10
Year,10
Genetic_Risk,101
Air_Pollution,101
Alcohol_Use,101
Smoking,101
Obesity_Level,101


The dataset contains diverse patient information, including multiple countries, cancer types, and stages. Most numerical features such as risk factors and treatment cost have a wide range of values, making them suitable for detailed analysis


In [ ]:
cancer_patients['Country_Region'].unique()
 #10 country regions comprising -> 'UK', 'China', 'Pakistan', 'Brazil', 'Germany', 'Canada', 'USA','India', 'Australia', 'Russia'

array(['UK', 'China', 'Pakistan', 'Brazil', 'Germany', 'Canada', 'USA',
       'India', 'Australia', 'Russia'], dtype=object)

In [ ]:
cancer_patients['Cancer_Type'].unique()
# 8 cancer Type comprising -> 'Lung', 'Leukemia', 'Breast', 'Colon', 'Skin', 'Cervical','Prostate', 'Liver'

array(['Lung', 'Leukemia', 'Breast', 'Colon', 'Skin', 'Cervical',
       'Prostate', 'Liver'], dtype=object)

In [ ]:
cancer_patients.isnull().sum()

,0
Patient_ID,0
Age,0
Gender,0
Country_Region,0
Year,0
Genetic_Risk,0
Air_Pollution,0
Alcohol_Use,0
Smoking,0
Obesity_Level,0


No missing values are present in the dataset, indicating clean data.

In [ ]:
cancer_patients.duplicated().sum()

np.int64(0)

No duplicate records were found

# Overall Observation:
# The dataset is clean and well-structured, with diverse features suitable for analysis of cancer severity and risk factors.

# **Exploratory Data Analysis**

1. Genetic_Risk VS Target_Severity_Score

In [4]:
cancer_patients.groupby('Genetic_Risk')['Target_Severity_Score'].mean()

,Target_Severity_Score
Genetic_Risk,
0.0,3.940528
0.1,3.971884
0.2,4.033265
0.3,4.076700
0.4,3.991635
...,...
9.6,5.883899
9.7,5.900262
9.8,5.887961


Observation --
“Severity score increases with genetic risk, indicating a positive relationship, although the increase is moderate.”

2. Smoking vs Target_Severity_Score

In [5]:
cancer_patients.groupby('Smoking')['Target_Severity_Score'].mean()

,Target_Severity_Score
Smoking,
0.0,3.918577
0.1,4.031002
0.2,4.017859
0.3,4.023889
0.4,4.056151
...,...
9.6,5.819329
9.7,5.909869
9.8,5.882200


“Severity score increases with higher smoking levels, indicating a positive relationship between smoking and cancer severity"

3. Air_Pollution vs Target_severity_score

In [16]:
cancer_patients.groupby('Air_Pollution')['Target_Severity_Score'].mean()


,Target_Severity_Score
Air_Pollution,
0.0,4.218103
0.1,4.247989
0.2,4.100674
0.3,4.213272
0.4,4.137987
...,...
9.6,5.629901
9.7,5.649459
9.8,5.667226


Observation:-
“Severity score increases with higher air pollution levels, indicating a moderate positive relationship.”

4. Alcohol_Use VS Target_severity_score

In [17]:
cancer_patients.groupby('Alcohol_Use')['Target_Severity_Score'].mean()

,Target_Severity_Score
Alcohol_Use,
0.0,4.263257
0.1,4.183066
0.2,4.248157
0.3,4.232451
0.4,4.314000
...,...
9.6,5.549940
9.7,5.641267
9.8,5.697495


Observation:-
“Severity score increases with higher alcohol consumption, indicating a moderate positive relationship between alcohol use and cancer severity.”

5. Obesity_Level vs Target_Severity_Score

In [18]:
cancer_patients.groupby('Obesity_Level')['Target_Severity_Score'].mean()

,Target_Severity_Score
Obesity_Level,
0.0,4.406860
0.1,4.471118
0.2,4.346216
0.3,4.493440
0.4,4.547797
...,...
9.6,5.448426
9.7,5.396203
9.8,5.533387


Observation:-
“Severity score increases with higher obesity levels, indicating a moderate positive relationship between obesity and cancer severity.”

6. Cancer_Type vs Target_Severity_Score

In [19]:
cancer_patients.groupby('Cancer_Type')['Target_Severity_Score'].mean()

,Target_Severity_Score
Cancer_Type,
Breast,4.947764
Cervical,4.951461
Colon,4.927520
Leukemia,4.967952
Liver,4.964380
Lung,4.937604
Prostate,4.943358
Skin,4.969957


Observation:-
“Severity scores are very similar across different cancer types, with only minor variations, suggesting that no cancer type is significantly more severe than others.”

7. Cancer_Stage vs Target_Severity_Score

In [20]:
cancer_patients.groupby('Cancer_Stage')['Target_Severity_Score'].mean()

,Target_Severity_Score
Cancer_Stage,
Stage 0,4.949560
Stage I,4.948788
Stage II,4.966201
Stage III,4.940766
Stage IV,4.950533


“Severity scores are very similar across different cancer stages, with only minor variations, suggesting that no cancer stage is significantly more severe than other.”

# Summary of EDA:-
1. Genetic risk shows a moderate positive relationship with severity score.  
2. Smoking has a strong positive impact on severity, indicating it is a major contributing factor.                                                       
3. Air pollution, alcohol use, and obesity levels show a moderate positive relationship with severity.                                                
4. Severity scores are fairly similar across different cancer types, indicating no significant variation.                                                    
5. Severity scores are also consistent across cancer stages, suggesting no major difference in severity between stages in this dataset.
